stdlib/public/core/StringHashable.swift

```diff
@@ -177,7 +191,9 @@ extension String : Hashable {
   /// your program. Do not save hash values to use during a future execution.
   @_inlineable // FIXME(sil-serialize-all)
   public var hashValue: Int {
-    return _guts._computeHashValue()
+    defer { _fixLifetime(self) }
```

Comment:
Is the `_fixLifetime` necessary here? If it is, we should audit for missing cases elsewhere; I started eliding it under the assumption that `self` lifetime is guaranteed.

Reply:
Self lifetime is not guaranteed, and technically these are always necessary. _LegacyStringCore violated this rule all over the place, and never blew up, but doing this is a good idea. @atrick was working on a dependent-lifetime improvement wherein we could say that some values/expressions/properties have lifetime dependent on self, and that would alleviate this.

Reply:
I think there's some confusion because self has a guaranteed calling convention, which is completely different from a fixed lifetime--it doesn't give you any lifetime guarantee. Currently, the best thing you can do is `defer { _fixLifetime }`. That's terrible for multiple reasons, but I haven't had any time to introduce a reasonable alternative.